src/lib/utilities.js

```diff
@@ -44,3 +45,64 @@ export function uuid() {
     s4() + '-' + s4() + s4() + s4();
 }
 
+
+// The following is all for global error handling.
+// We keep the globalError value on top of AppController's 
+// this.state.error because this.setState is asynchronous
+// while this variable will be synchronous and can therefore
+// always be checked
+let globalError = null;
+let appControllerMounted = false;
+
+// These will be the functions with 'this' bound to AppController
+// So they can change the state of it globally
+let boundSetGlobalError, boundResetGlobalError;
+
+// Remember to call this function with the correct 'this' value
+export function setContextForGlobalErrorFunctions() {
```

Comment:
This bind thing is confusing. Just pass the context in as an argument directly, much simpler.